virttest/vt_utils/block.py

```diff
@@ -0,0 +1,204 @@
+                         r'(?P<size>\d+\w+)\s+(?P<type>\w+)' % part_num)
+    return pattern.search(process.run(parted_cmd).stdout_text, re.M).groupdict()
+
+
+def resize_partition(partition, size):
+    """
+    Resize partition.
+    Note: not support gpt type resize for linux guest.
+
+    :param partition: partition that to be shrunk, like /dev/sdb1.
+    :param size: resize partition to size, unit is B.
+    """
+    mountpoint, fstype = filesystem.get_mpoint_fstype(partition)
+    if filesystem.is_mounted(partition, dst=mountpoint, fstype=fstype):
+        filesystem.umount(partition, mountpoint, fstype=fstype)
+
+    # FIXME: if nvme device need support this function.
+    dev_name, part_num = re.match(r'(/dev/.*)(\d+)', partition).groups()
+    parted_cmd = 'parted -s %s print' % dev_name
+    part_attrs = get_partition_attrs(partition)
+    start_size = int(utils_numeric.normalize_data_size(part_attrs['start'],
+                                                       'B').split('.')[0])
+    end_size = (int(utils_numeric.normalize_data_size(size, 'B').split('.')[0])
+                - start_size)
+    process.system(' '.join((parted_cmd, 'rm %s' % part_num)))
+    resizepart_cmd = ' '.join((parted_cmd, 'unit B mkpart {0} {1} {2}'))
+    process.system(resizepart_cmd.format(part_attrs['type'], start_size,
+                                         end_size))
+    process.system('partprobe %s' % dev_name)
+
+    if fstype == 'xfs':
+        process.system('xfs_repair -n %s' % partition)
+    elif fstype.startswith("ext"):
+        process.system('e2fsck -f %s' % partition)
+    else:
+        raise NotImplementedError("%s is not supported!" % fstype)
+
+
+def get_disk_size(did):
+    """
+    Get disk size.
+
+    :param did: disk kname. e.g. 'sdb', 'sdc'
+    :return: disk size.
+    """
+    disks_info = get_disks(partition=True)
+    disk_size = disks_info['%s' % did][1]
+    return int(utils_numeric.normalize_data_size(disk_size, 'B').split('.')[0])
+
+
+def get_partions_list():
+    """
+    Get all partition lists.
+    """
+    parts_cmd = "cat /proc/partitions"
+    parts_out = process.run(parts_cmd).stdout_text
+    parts = []
+    if parts_out:
+        for line in parts_out.rsplit("\n"):
+            if line.startswith("major") or line == "":
+                continue
+            parts_line = line.rsplit()
+            if len(parts_line) == 4:
+                parts.append(parts_line[3])
+    return parts
+
+
+def get_disk_by_serial(serial_str):
+    """
+    Get disk by serial in host.
+
+    :param serial_str: ID_SERIAL of disk, string value
+    :return: Disk name if find one with serial_str, else None
+    """
+    parts_list = get_partions_list()
+    for disk in parts_list:
+        cmd = ("udevadm info --query=all --name=/dev/{} | grep ID_SERIAL={}"
+               .format(disk, serial_str))
+        status = process.run(cmd, shell=True, ignore_status=True).exit_status
+        if not status:
+            return disk
```

Comment:
Do we want to return None explicitly in the end although the function will return None implicitly

Reply:
Both are okay for me.